java/code/src/com/redhat/rhn/domain/contentmgmt/ProjectSource.java

```diff
@@ -41,8 +45,107 @@
 public abstract class ProjectSource {
 
     private Long id;
+    private State state;
     private ContentProject contentProject;
 
+    /**
+     * State of the Source
```

Comment:
Just to be nice with developers new to this feature, I would add a small description like:
```
ATTACHED, // softly added to the Project but not yet part of the Build output
DETACHED, // softly removed from the Project will be really removed after the next Project Build
BUILT, // part of the latest Build
```

Reply:
Good idea. done.